SDK/MobileCenter/Microsoft.Azure.Mobile.Windows.Shared/MobileCenter.cs

```diff
@@ -292,32 +296,42 @@ internal void StartInstance(params Type[] services)
                 }
                 try
                 {
-                    var serviceInstance = serviceType.GetRuntimeProperty("Instance")?.GetValue(null) as IMobileCenterService;
-                    StartService(serviceInstance);
+                    var serviceInstance =
+                        (IMobileCenterService) serviceType.GetRuntimeProperty("Instance")?.GetValue(null);
+
+                    if (StartService(serviceInstance))
+                        startedServiceNames.Add(serviceInstance.ServiceName);
                 }
                 catch (MobileCenterException ex)
                 {
                     MobileCenterLog.Warn(MobileCenterLog.LogTag, $"Failed to start service '{serviceType.Name}'; skipping it.", ex);
                 }
             }
+
+            if (startedServiceNames.Count > 0)
+            {
+                StartServiceLog serviceLog = new StartServiceLog();
+                serviceLog.Services = startedServiceNames;
+                _channelGroup.Enqueue( serviceLog );
+            }
         }
 
-        private void StartService(IMobileCenterService service)
+        private bool StartService(IMobileCenterService service)
```

Comment:
Or you can still use a list, right? If an exception is thrown in start service, it won't be added because the exception will be thrown before the service can be added to the list. Instead of logging a message in the StartService method, just throw a MobileCenterException that contains that message